### PR TITLE
Benchmarks: Add Benchmark - Add GPT2 model benchmark.

### PR DESCRIPTION
**Pull request by guoshzhao**
**Description**
Add GPT2 related benchmarks

**Major Revision**
- Add PytorchGPT2 benchmark class.
- Register 'pytorch-gpt2-small', 'pytorch-gpt2-medium', 'pytorch-gpt2-large' and 'pytorch-gpt2-xl' benchmark.

**Minor Revision**
- Add tests for gpt2 benchmark.
- Add example for gpt2-large benchmark.

### CODE REVIEW

File: examples/benchmarks/pytorch_gpt2_large.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+from superbench.benchmarks import Platform, Framework, BenchmarkRegistry
+from superbench.common.utils import logger
+
+if __name__ == '__main__':
+    # Create context for gpt2-large benchmark and run it for 120 * 2 seconds.
+    context = BenchmarkRegistry.create_benchmark_context(
+        'gpt2-large',
+        platform=Platform.CUDA,
+        parameters='--batch_size 1 --duration 120 --seq_len 128 --precision float32 --run_count 2',
```
Comment (cp5555) [inline examples/benchmarks/pytorch_gpt2_large.py:19]: What's our suggested sequence length for testing?

File: superbench/benchmarks/model_benchmarks/pytorch_gpt2.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        super().__init__(name, parameters)
+        self._config = None
+        self._supported_precision = [Precision.FLOAT32, Precision.FLOAT16]
+        self._optimizer_type = Optimizer.ADAMW
+        self._loss_fn = torch.nn.CrossEntropyLoss()
+
+    def add_parser_arguments(self):
+        """Add the GPT2-specified arguments.
+
+        BERT model reference: https://huggingface.co/transformers/model_doc/gpt2.html
```
Comment (cp5555) [inline superbench/benchmarks/model_benchmarks/pytorch_gpt2.py:65]: GPT-2 model reference

File: superbench/benchmarks/model_benchmarks/pytorch_gpt2.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        """
+        duration = []
+        curr_step = 0
+        with torch.no_grad():
+            self._model.eval()
+            while True:
+                for idx, sample in enumerate(self._dataloader):
+                    torch.cuda.synchronize()
+                    start = time.time()
+                    sample = sample.cuda()
```
Comment (cp5555) [inline superbench/benchmarks/model_benchmarks/pytorch_gpt2.py:170]: Do we need to check `if self._gpu_available:`

File: examples/benchmarks/pytorch_gpt2_large.py (hunk 1 above)
Comment (guoshzhao) [reply examples/benchmarks/pytorch_gpt2_large.py:19]: I think for example, 256 or 512 may be better. But considering the memory footprint (we have set batch_size as 1), 128 should also be ok.

File: superbench/benchmarks/model_benchmarks/pytorch_gpt2.py (hunk 2 above)
Comment (guoshzhao) [reply superbench/benchmarks/model_benchmarks/pytorch_gpt2.py:65]: Good catch.

File: superbench/benchmarks/model_benchmarks/pytorch_gpt2.py (hunk 3 above)
Comment (guoshzhao) [reply superbench/benchmarks/model_benchmarks/pytorch_gpt2.py:170]: Good catch.

File: examples/benchmarks/pytorch_gpt2_large.py (hunk 1 above)
Comment (cp5555) [reply examples/benchmarks/pytorch_gpt2_large.py:19]: That's great!

**Review by cp5555 — APPROVED** [review]

File: tests/benchmarks/model_benchmarks/test_pytorch_gpt2.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Microsoft Corporation.
```
Comment (cp5555) [inline tests/benchmarks/model_benchmarks/test_pytorch_gpt2.py:1]: @guoshzhao After the discussion, we should keep the code coverage on patch is higher than 80%

**Review by abuccts — APPROVED** [review]

File: tests/benchmarks/model_benchmarks/test_pytorch_gpt2.py (hunk 4 above)
Comment (guoshzhao) [reply tests/benchmarks/model_benchmarks/test_pytorch_gpt2.py:1]: Done. Currently `codecov/patch — 94.68% of diff hit (target 80.00%)`

**Pull request merged**